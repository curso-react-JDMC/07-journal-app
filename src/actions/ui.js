import { types } from "../types/types";


export const uiSetError = (error)=> ({
    type: types.uiSetError,
    payload:error
})


export const uiRemoveError = ()=> ({
    type: types.uiRemoveError
})

export const uiStartLoading = ()=> ({
    type: types.uiStartLoading,
})

export const uiFinishLoading = ()=> ({
    type: types.uiFinishLoading,
})