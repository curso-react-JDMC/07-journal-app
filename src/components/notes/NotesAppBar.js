import React from 'react'

export const NotesAppBar = () => {
  return (
    <div className="notes__appbar">
        <span>23 septiembre 2022</span>
        <div>
            <button className="btn ">Picture</button>
            <button className="btn ">Save</button>
        </div>
    </div>
  )
}
