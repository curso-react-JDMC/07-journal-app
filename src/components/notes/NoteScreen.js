import React from "react";
import { NotesAppBar } from "./NotesAppBar";

export const NoteScreen = () => {
  return (
    <div className="notes__main-content">
      <NotesAppBar />
      <div className="notes__content">
        <input
          type="text"
          placeholder="Some awesome title"
          className="notes__title-input"
          autoComplete="off"
        />
        <textarea
          cols="30"
          rows="10"
          placeholder="what happened today"
          className="notes__text-area"
        ></textarea>
        <div className="notes__image">
            <img src="https://images.unsplash.com/photo-1453728013993-6d66e9c9123a?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8Mnx8dmlld3xlbnwwfHwwfHw%3D&w=1000&q=80" alt="imagen" />
        </div>
      </div>
    </div>
  );
};
