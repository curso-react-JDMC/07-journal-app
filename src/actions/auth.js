import { firebase, googleAuthProvider } from "../firebase/firebaseConfig";
import { types } from "../types/types";
import { uiFinishLoading, uiStartLoading } from "./ui";
import Swal from 'sweetalert2';
import { notesLogout } from "./notes";

export const startLoginEmail = (email, password) => {
  return (dispatch) => {
    dispatch(uiStartLoading());
    firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(({ user }) => {
        //console.log(user);
        dispatch(login(user.uid, user.displayName))
        dispatch(uiFinishLoading());
      }).catch(err => {
        console.log(err);
        dispatch(uiFinishLoading());
        Swal.fire('Error',err.message, 'error');
      })
  };
};

export const startRegisterEmailPass = (name, email, password) => {
  return (dispatch) => {
    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(async ({ user }) => {
        await user.updateProfile({ displayName: name });
        dispatch(login(user.uid, user.displayName));
      })
      .catch((e) => {
        console.log(e);
      });
  };
};

export const startGoogleLogin = () => {
  return (dispatch) => {
    firebase
      .auth()
      .signInWithPopup(googleAuthProvider)
      .then(({ user }) => {
        dispatch(login(user.uid, user.displayName));
      });
  };
};

export const login = (uid, displayName) => ({
  type: types.login,
  payload: {
    uid,
    displayName,
  },
});

export const logout = () => ({
  type: types.logout
});

export const startLogout = () => {
  return async (dispatch) => {
    await firebase.auth().signOut();
    dispatch(logout());
    dispatch(notesLogout());
  }
};