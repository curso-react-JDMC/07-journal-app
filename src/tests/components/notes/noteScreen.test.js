import { mount } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { activeNote } from "../../../actions/notes";
import { Sidebar } from "../../../components/journal/Sidebar";
import { NoteScreen } from "../../../components/notes/NoteScreen";

jest.mock("../../../actions/notes", () => ({
  activeNote: jest.fn(),
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initState = {
  auth: {
    uid: "1",
    name: "juan david",
  },
  ui: {
    loading: false,
    msgError: null,
  },
  notes: {
    active: {
      id: "stG0b3GuokUXPJO8ZcrN",
      title: "title",
      body: "mundo",
    },
    notes: [],
  },
};
let store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
  <Provider store={store}>
    <NoteScreen />
  </Provider>
);

describe("pruebas en noteScreen", () => {
  test("debe mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe disparar el activeNore", () => {
    wrapper.find('input[name="title"]').simulate("change", {
      target: {
        name: "title",
        value: "Hola",
      },
    });

    expect(activeNote).toHaveBeenLastCalledWith("stG0b3GuokUXPJO8ZcrN", {
      body: "mundo",
      id: "stG0b3GuokUXPJO8ZcrN",
      title: "Hola",
    });
  });
});
