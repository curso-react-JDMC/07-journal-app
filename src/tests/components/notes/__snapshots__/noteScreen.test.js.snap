// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en noteScreen debe mostrarse correctamente 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      23 septiembre 2022
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn "
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Some awesome title"
      type="text"
      value="title"
    />
    <textarea
      className="notes__text-area"
      cols="30"
      name="body"
      onChange={[Function]}
      placeholder="what happened today"
      rows="10"
      value="mundo"
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
